# [LAB 08] 1. EDA 시작하기
## #연습문제
![alt text](image.png)

In [2]:
from hossam import load_data
from helpers import *

📦 아이티윌 이광호 강사가 제작한 라이브러리를 사용중입니다.
📚 자세한 사용 방법은 https://py.hossam.kr 을 참고하세요.
📧 Email: leekh4232@gmail.com
🎬 Youtube: https://www.youtube.com/@hossam-codingclub
📝 Blog: https://blog.hossam.kr/
🔖 Version: 0.5.19


In [3]:
origin = load_data('insurance_qtcheck')
print(origin.shape)
print(origin.head())

📚 보험 비용 데이터셋의 품질 검사 완료 버전 (출처: 자체 정제)
(1337, 7)
   age     sex    bmi  children smoker     region   charges
0   19  female 27.900         0    yes  southwest 16884.924
1   18    male 33.770         1     no  southeast  1725.552
2   28    male 33.000         3     no  southeast  4449.462
3   33    male 22.705         0     no  northwest 21984.471
4   32    male 28.880         0     no  northwest  3866.855


- 보험 비용 데이터 셋의 행과 열은 각각 1337, 7이며 컬럼은 age, sex, bmi, children, smoker, region, charges이고 목표변수는 보험 비용인 charges 이다.

In [4]:
desc = load_data('insurance_qtcheck_desc')
desc.T

📚 보험 비용 데이터셋의 연속형 변수에 대한 기술 통계량 (출처: 자체 작업)


,age,bmi,children,charges
count,1337.000,1337.000,1337.000,1337.000
mean,39.222,30.663,1.096,13279.121
std,14.044,6.100,1.206,12110.360
min,18.000,15.960,0.000,1121.874
25%,27.000,26.290,0.000,4746.344
50%,39.000,30.400,1.000,9386.161
75%,51.000,34.700,2.000,16657.717
max,64.000,53.130,5.000,63770.428
rel_diff,0.006,0.009,0.096,0.415
rdiff_flag,similar,similar,similar,diff


- children, charges가 로그 변환 후보이다. 둘다 오른쪽 꼬리 분포를 보인다.

In [5]:
cat_desc = load_data('insurance_qtcheck_category_desc')
cat_desc

📚 보험 비용 데이터셋의 범주형 변수에 대한 기술 통계량 (출처: 자체 작업)


,Unnamed: 0,sex,smoker,region
0,count,1337,1337,1337
1,unique,2,2,4
2,top,male,no,southeast
3,freq,675,1063,364


- sex, smoker, region이 명목형 변수이며 이중 smoker의 freq이 1063으로 한쪽으로 쏠렸다고 볼 수 있다. 심한 불균형이나 흡연 여부는 보험 비용에 영향을 줄 수 있는 변수이기 때문에 버리지 말고 검정으로 확인하는 것이 필요할 것으로 보인다. 

In [6]:
origin.info()

<class 'pandas.DataFrame'>
RangeIndex: 1337 entries, 0 to 1336
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype   
---  ------    --------------  -----   
 0   age       1337 non-null   int64   
 1   sex       1337 non-null   category
 2   bmi       1337 non-null   float64 
 3   children  1337 non-null   int64   
 4   smoker    1337 non-null   category
 5   region    1337 non-null   category
 6   charges   1337 non-null   float64 
dtypes: category(3), float64(2), int64(2)
memory usage: 45.9 KB


- 명목형 변수는 이미 범주형 타입으로 잘 지정되어 있으므로 별도의 타입 변환은 필요 없다.

In [ ]:
목표변수 = 'charges'

연속형_설명변수 = ['age', 'bmi', 'children']
범주형_설명변수 = ['sex', 'smoker', 'region']

print(목표변수)
print(연속형_설명변수)
print(범주형_설명변수)

charges
['age', 'bmi', 'children']
['sex', 'smoker', 'region']
